### PR TITLE
Implemented saveToMemory and saveImageToMemory.

### PR DESCRIPTION
**Pull request by JonnyPtn**
Follow up from #1597. Just fixed the merge conflicts and tested

### CODE REVIEW

File: include/SFML/Graphics/Image.hpp [hunk 1]
```diff
@@ -155,6 +155,26 @@ class SFML_GRAPHICS_API Image
...
+    ///
+    /// The format of the image must be specified.
+    /// The supported image formats are bmp, png, tga and jpg.
+    /// This function fails if the image is empty, or if
+    /// the format was invalid.
+    ///
+    /// \param output Buffer to fill with encoded data
+    /// \param format Encoding format to use
+    /// \param output Buffer to fill with encoded data
+    /// \param format Encoding format to use
```
Comment (eXpl0it3r) [inline include/SFML/Graphics/Image.hpp:169]: Duplicated `\param` comments

Comment (JonnyPtn) [reply include/SFML/Graphics/Image.hpp]: Good spot, must have messed it up resolving conflicts - fixed now

**Comment by eXpl0it3r** [comment]
Wrote a bit a longer test example that checks:
- Writing data in memory to file (to check integrity)
- Load data from memory into another `sf::Image` instance

Seems to work nicely.

```cpp
#include <SFML/Graphics.hpp>

#include <fstream>
#include <vector>

int main()
{
    // Load or create any image.
    sf::Image image;
    image.loadFromFile("test.png");
    image.saveToFile("test_file.png");

    // Create a buffer and encode image to it
    std::vector<sf::Uint8> content;
    image.saveToMemory(content, "png");

    sf::Image image2;
    image2.loadFromMemory(content.data(), content.size());

    auto file = std::fstream{ "test_memory.png", std::ios::out | std::ios::binary };

    for (auto c : content)
    {
        file << c;
    }

    file.close();

    auto window = sf::RenderWindow{ {1280, 720}, "SaveToMemory Test" };
    window.setFramerateLimit(30);

    auto texture = sf::Texture{};
    texture.loadFromImage(image);
    auto sprite = sf::Sprite{ texture };

    auto texture2 = sf::Texture{};
    texture2.loadFromImage(image2);
    auto sprite2 = sf::Sprite{ texture };
    sprite2.setPosition({ 100.f, 100.f });

    while(window.isOpen())
    {
        for (auto event = sf::Event{}; window.pollEvent(event);)
        {
            if (event.type == sf::Event::Closed)
            {
                window.close();
            }
        }

        window.clear();
        window.draw(sprite);
        window.draw(sprite2);
        window.display();
    }
}
```

**Comment by Bromeon** [comment]
> Wrote a bit a longer test example that checks:

Nice! Should we maybe add such code examples to the unit tests?
Or, to save time, throw them in a central place (Wiki or so) and properly integrate them later?

**Comment by binary1248** [comment]
SFMLBuildMaster: Build this please.

**Pull request merged**